dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs

```diff
@@ -710,26 +726,34 @@ private static (Func<KernelFunction, Kernel, KernelArguments, CancellationToken,
 
             object? Process(object? value)
             {
-                if (!type.IsAssignableFrom(value?.GetType()))
+                if (type.IsAssignableFrom(value?.GetType()))
                 {
-                    if (converter is not null)
+                    return value;
+                }
+
+                if (converter is not null && value is not JsonElement or JsonDocument or JsonNode)
```

Comment:
I wonder whether you want to use converters for JsonDocument and JsonNode types? or it's a bug and the expression should be `if (converter is not null && value is not (JsonElement or JsonDocument or JsonNode))` instead?